ext/dom/tests/DOMElement_className.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+DOMElement::className
+--EXTENSIONS--
+dom
+--FILE--
+<?php
+
+class MyStringable {
+    public function __toString(): string {
+        throw new Exception("foo");
+    }
+}
+
+$dom = new DOMDocument();
+$dom->loadXML('<html/>');
+
+var_dump($dom->documentElement->className);
+$dom->documentElement->className = "hello & world<>";
+var_dump($dom->documentElement->className);
+$dom->documentElement->className = "";
+var_dump($dom->documentElement->className);
+$dom->documentElement->className = "é";
+var_dump($dom->documentElement->className);
+$dom->documentElement->className = "\0";
```

Comment:
This might need a note in UPGRADING and the docs that this is not binary safe (i.e. maybe add an example of a string like ``"ab\0de"``

Reply:
Actually, shouldn't such invalid names possibly throw a DOM Exception like it is done in #11696 ?

Reply:
Unfortunately, these characters _are_ allowed in the value of the attribute, just not in the name.